roles/common/tasks/config_validations.yml

```diff
@@ -206,3 +206,47 @@
     fail_msg: "OAuth is not supported on FIPS enabled cluster yet. Please disable FIPS or OAuth to setup the cluster."
   when: auth_mode == 'oauth' or auth_mode == 'ldap_with_oauth'
   tags: validate
+
+- assert:
+    that:
+      - sasl_protocol.replace(' ', '').split(',') | length < 3
+    fail_msg: "sasl_protocol can have support for at most two values."
+  when: sasl_protocol is defined
+  tags: validate
+
+- assert:
+    that:
+      - sasl_protocol.replace(' ', '').split(',')[0] in ['plain', 'scram', 'scram256', 'kerberos', 'oauth', 'none']
+    fail_msg: "sasl_protocol supports ['plain', 'scram', 'scram256', 'kerberos', 'oauth', 'none'] when providing single value."
+  when: sasl_protocol is defined and sasl_protocol.replace(' ', '').split(',') | length == 1
+  tags: validate
+
+- assert:
+    that:
+      - item in ['plain', 'scram', 'scram256', 'kerberos']
+    fail_msg: "sasl_protocol can have support for two values from the list ['plain', 'scram', 'scram256', 'kerberos']"
+  loop: "{{sasl_protocol.replace(' ', '').split(',')}}"
+  when: sasl_protocol is defined and sasl_protocol.replace(' ', '').split(',') | length == 2
+  tags: validate
+
+- assert:
+    that:
+      - kafka_controller_sasl_protocol.replace(' ', '').split(',') | length < 3
+    fail_msg: "kafka_controller_sasl_protocol can have support for at most two values."
+  when: kafka_controller_sasl_protocol is defined
+  tags: validate
+
+- assert:
+    that:
+      - kafka_controller_sasl_protocol.replace(' ', '').split(',')[0] in ['plain', 'scram', 'scram256', 'kerberos', 'oauth', 'none']
```

Comment:
how are we supporting scram as kafka controller single protocol here?

Reply:
Was handling it in scram changes, but removed from here as well.

Reply:
i don't think we should support that either, there are way too many workaround to make scram work in controller. Please restrict the support for inter-controller to non-scram, irrespective of the number of controllers

Reply:
added